xlrd/xldate.py

```diff
@@ -130,6 +131,76 @@ def xldate_as_datetime(xldate, datemode):
     return epoch + datetime.timedelta(days, seconds, 0, milliseconds)
 
 
+##
+# Convert an Excel date/time in ISO date format into a serial number.
+#
+# @param xldate The Excel number
+# @param datemode 0: 1900-based, 1: 1904-based.
+#
+# @return an Excel serial date number.
+#
+def xldate_strict_to_number(xldate, datemode):
+    """
+    Convert a Excel ISO date (from Excel 2013 in strict Open XML mode)
+    to a serial date and time.
+
+    """
+
+    if datemode:
+        # Excel for Mac date epoch.
+        epoch = epoch_1904
+    else:
+        # Default Excel epoch.
+        epoch = epoch_1900_minus_1
+
+    if re.search("T", xldate):
+        # The Excel date is in a date and time format.
+        m = re.match(r'(\d\d\d\d)-(\d\d)-(\d\d)T(\d\d):(\d\d):(\d\d)(.\d+)',
+                     xldate)
```

Comment:
Do we really need a date, though? datetime.strptime shoulld be able to handle this, right?